app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/dashboard.tsx

```diff
@@ -321,12 +322,76 @@ function Dashboard(props: DashboardProps) {
         })()
     }, [accessToken, directoryId])
 
+    const getNewProjectName = (templateName: string | undefined): string => {
+        const projectNameTemplateStart = templateName ? `${templateName}_` : 'New_Project_'
+        const projectNameTemplate = new RegExp(`^${projectNameTemplateStart}(?<projectIndex>\\d+)$`)
+        let lastProjectIndex = 0
+        projectAssets.forEach(projectItem => {
+            let projectIndex: string | number | undefined = projectNameTemplate.exec(
+                projectItem.title
+            )?.groups?.projectIndex
+            if (!projectIndex) {
+                return
+            }
+
+            projectIndex = parseInt(projectIndex, 10)
+            if (projectIndex > lastProjectIndex) {
+                lastProjectIndex = projectIndex
+            }
+        })
+        lastProjectIndex++
+
+        return projectNameTemplateStart + lastProjectIndex.toString()
+    }
+    const handleCreateProject = async (templateName: string | undefined) => {
+        const projectName = getNewProjectName(templateName)
+        switch (platform) {
+            case platformModule.Platform.cloud: {
+                const body: backend.CreateProjectRequestBody = {
+                    projectName,
+                }
+                if (templateName) {
+                    body.projectTemplateName = templateName.replace(/_/g, '').toLocaleLowerCase()
+                }
+                const project = await backendService.createProject(body)
+                setProjectAssets(p => [
+                    ...p,
+                    {
+                        type: backend.AssetType.project,
+                        title: project.name,
+                        id: project.projectId,
+                        parentId: '',
+                    },
+                ])
+                break
+            }
+            case platformModule.Platform.desktop: {
+                const body: projectManagerModule.CreateProjectParams = {
+                    name: newtype.asNewtype(projectName),
```

Comment:
prefer explicitly passing the specific type, so that readers will be able to tell that it's being cast to the correct type